src/Notifications/RefreshToken/DiscordRefreshTokenNotification.php

```diff
@@ -17,4 +21,28 @@ public static function hasPersonalNotification() : bool
     {
         return true;
     }
+
+    public function via($notifiable)
+    {
+        array_push($this->tags, is_null($notifiable->group_id) ? 'to channel' : 'private to: ' . $this->getMainCharacter(Group::find($notifiable->group_id))->name);
```

Comment:
why are you not using `main_character` property which you've set in the constructor ?

Reply:
So for this Notification we must consider two different `main_character`s. 

One from the constructor is the `main_character` of the character that had deleted its `refresh_token`. The other is the `main_character` of the recipient.